lib/requests/diaryFoods.js

```diff
@@ -0,0 +1,61 @@
+const getDiaryFoods = () => {
+  fetch(`https://ivmh-qs-api.herokuapp.com/api/v1/foods`)
+    .then(response => handleResponse(response))
+    .then(foods => getEachDiaryFood(foods))
+    .catch(error => console.error({ error }))
+}
+
+const handleResponse = (response) => {
+  return response.json()
+    .then(json => {
+      if (!response.ok) {
+        const error = {
+          status: response.status,
+          statusTest: response.statusText,
+          json
+        }
+        return Promise.reject(error)
+      }
+      return json
+    })
+}
+
+const getEachDiaryFood = (foods) => {
+  return foods.forEach(food => {
+    renderDiaryFood(food)
+  })
+}
+
+const renderDiaryFood = (food) => {
+  $('#food-table-info').prepend(
+   `<article class="food-item-${food.id}" id="food-item-row" data="food-${food.id}">
+     <div class="checkbox-container">
+      <input id="food-item-${food.id}" type="checkbox" class="food-item-checkbox">
+     </div>
+      <p class="food-item-name" contenteditable="true">${food.name}</p>
+      <p class="food-item-calories" contenteditable="true">${food.calories}</p>
+    </article>`
+  )
+}
+
+const updateDiaryFood = (id) => {
```

Comment:
Looking back at the stories, is a user expected to be updating a food item from the diary? As I read it it seems as though the story is saying that even when a user updates a food item on the foods.html chart, those changes should persist in the diary. Which would make sense because editing a food item attribute from the chart in foods.html actually updates the food via update fetch request to the API.

Of course the food would be updated in the diary since it 'gets' all the food via request to the same API.

My question is - is the story calling for a user being able to update a food item in BOTH the foods.html page AND the index.html food diary?

Reply:
The initial card for making food editable doesn't specify if the foods should be editable on only foods.html, so I took it to mean that a food should be editable on both pages.

https://www.pivotaltracker.com/story/show/154615166
"If I click the Name or Calories field for any given food, it will become an input field containing the current values for that food"

If the foods should only be editable on the foods.html page, I can make that change.

Reply:
Sounds good. I think we could merge this for now and continue working and if we find that the story is different from what we have in our code then we can make those changes.

Shall I merge for now or do you prefer to wait on their feedback? @marshallhouston

Reply:
@izaxveta @marshallhouston - just making the food editable on the foods page is sufficient!

Y'all are killing it!!